Rendering of the 404 Not Found page breaks because renderingMode is not available

@mhsdesign It seems this https://github.com/neos/neos-development-collection/pull/4733 changes breaks rendering of the 404 Not Found page in Neos Demo (Neos.Demo:Document.NotFoundPage)

Exception:
```
Fatal error: Uncaught AssertionError: assert($renderingMode instanceof RenderingMode) in /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Neos_Fusion_ConvertUrisImplementation.php:128 Stack trace: #0 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Neos_Fusion_ConvertUrisImplementation.php(128): assert(false, 'assert($renderi...') #1 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(527): Neos\Neos\Fusion\ConvertUrisImplementation_Original->evaluate() #2 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(479): Neos\Fusion\Core\Runtime_Original->evaluateObjectOrRetrieveFromCache(Object(Neos\Neos\Fusion\ConvertUrisImplementation), 'header<Neos.Dem...', Array, Array) #3 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(868): Neos\Fusion\Core\Runtime_Original->evaluate('header<Neos.Dem...', Object(Neos\Fusion\FusionObjects\JoinImplementation), 'Exception') #4 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(534): Neos\Fusion\Core\Runtime_Original->evaluateProcessors('<svg class="hid...', Array, 'header<Neos.Dem...', Object(Neos\Fusion\FusionObjects\JoinImplementation)) #5 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(479): Neos\Fusion\Core\Runtime_Original->evaluateObjectOrRetrieveFromCache(Object(Neos\Fusion\FusionObjects\JoinImplementation), 'header<Neos.Dem...', Array, Array) #6 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(314): Neos\Fusion\Core\Runtime_Original->evaluate('header<Neos.Dem...', NULL, 'Exception') #7 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_FusionObjects_ComponentImplementation.php(96): Neos\Fusion\Core\Runtime_Original->render('header<Neos.Dem...') #8 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_FusionObjects_ComponentImplementation.php(46): Neos\Fusion\FusionObjects\ComponentImplementation_Original->render(Array) #9 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(527): Neos\Fusion\FusionObjects\ComponentImplementation_Original->evaluate() #10 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(479): Neos\Fusion\Core\Runtime_Original->evaluateObjectOrRetrieveFromCache(Object(Neos\Fusion\FusionObjects\ComponentImplementation), 'header', Array, Array) #11 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(314): Neos\Fusion\Core\Runtime_Original->evaluate('header', NULL, 'Exception') #12 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_FusionObjects_RendererImplementation.php(58): Neos\Fusion\Core\Runtime_Original->render('header') #13 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(527): Neos\Fusion\FusionObjects\RendererImplementation_Original->evaluate() #14 /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Fusion_Core_Runtime.php(479): Neos\Fusion\Core\Runtime_Original->evaluateObjectOrRetrieveFromCache(Object(Neos\Fusion\FusionObjects\RendererImplementation), 'root<Neos.Fusio...', Array, Array) #15 /var/www/html/Packages/Neos/Neos.Fusion/Classes/FusionObjects/Helpers/LazyProps.php(69): Neos\Fusion\Core\Runtime_Original->evaluate('root<Neos.Fusio...', Object(Neos\Fusion\FusionObjects\ComponentImplementation)) #16 /var/www/html/Packages/Neos/Neos.Fusion/Classes/FusionObjects/Helpers/LazyProps.php(93): Neos\Fusion\FusionObjects\Helpers\LazyProps->offsetGet('header') #17 [internal function]: Neos\Fusion\FusionObjects\Helpers\LazyProps->current() #18 /var/www/html/Packages/Neos/Neos.Fusion/Classes/FusionObjects/Helpers/LazyProps.php(118): iterator_to_array(Object(Neos\Fusion\FusionObjects\Helpers\LazyProps)) #19 [internal function]: Neos\Fusion\FusionObjects\Helpers\LazyProps->jsonSerialize() #20 /var/www/html/Packages/Framework/Neos.Flow/Classes/Error/Debugger.php(478): json_encode(Object(Neos\Fusion\FusionObjects\Helpers\LazyProps), 512, 1) #21 /var/www/html/Packages/Framework/Neos.Flow/Classes/Error/Debugger.php(341): Neos\Flow\Error\Debugger::getObjectSnippetPlaintext(Object(Neos\Fusion\FusionObjects\Helpers\LazyProps)) #22 /var/www/html/Packages/Framework/Neos.Flow/Classes/Error/DebugExceptionHandler.php(130): Neos\Flow\Error\Debugger::getBacktraceCode(Array) #23 /var/www/html/Packages/Framework/Neos.Flow/Classes/Error/DebugExceptionHandler.php(81): Neos\Flow\Error\DebugExceptionHandler->renderStatically(404, Object(AssertionError)) #24 /var/www/html/Packages/Framework/Neos.Flow/Classes/Error/AbstractExceptionHandler.php(128): Neos\Flow\Error\DebugExceptionHandler->echoExceptionWeb(Object(Neos\Flow\Mvc\Controller\Exception\InvalidControllerException)) #25 [internal function]: Neos\Flow\Error\AbstractExceptionHandler->handleException(Object(Neos\Flow\Mvc\Controller\Exception\InvalidControllerException)) #26 {main} thrown in /var/www/html/Data/Temporary/Development/SubContextDdev/Cache/Code/Flow_Object_Classes/Neos_Neos_Fusion_ConvertUrisImplementation.php on line 128
```

Is this a Neos or Neos.Demo issue?

_Originally posted by @dlubitz in https://github.com/neos/neos-development-collection/issues/4733#issuecomment-1872192592_

--- a/Neos.Neos/Classes/View/FusionExceptionView.php
+++ b/Neos.Neos/Classes/View/FusionExceptionView.php
@@ -205,7 +205,7 @@ class FusionExceptionView extends AbstractView
 
             $fusionGlobals = FusionGlobals::fromArray([
                 'request' => $controllerContext->getRequest(),
-                'renderingModeName' => RenderingMode::FRONTEND
+                'renderingMode' => RenderingMode::createFrontend()
             ]);
             $this->fusionRuntime = $this->runtimeFactory->createFromConfiguration(
                 $fusionConfiguration,

